### PR TITLE
Revert Go version to 1.22.0

### PR DESCRIPTION
**Pull request by lucacome**
### Proposed changes

This is a library so it will be compiled with the version specified in the project that imports it. This also is treated by Go as a minimum suggested version and we should only increase it if we need new features from the language. Bumping this to the latest version only means that the projects importing it will be forced to update to the same version.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/telemetry-exporter/blob/main/CONTRIBUTING.md)
      guide
- [ ] I have proven my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [x] I have ensured the README is up to date
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch on my own fork

### CODE REVIEW

**Review by sjberman — APPROVED** [review]

**Comment by danielnginx** [comment]
That fix security issues we should not revert it. https://go.dev/doc/devel/release#go1.22.4

**Comment by MichaelMcAleer** [comment]
I second the comment from @danielnginx , we should not roll back here

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/nginxinc/telemetry-exporter
 
-go 1.22.4
```
Comment (MichaelMcAleer) [inline go.mod:3]: I agree with @danielnginx this should not be downgraded, the purpose of this bump is to consume Go stdlib vuln fixes.

Please reach out to Nginx security team with any questions.

**Review by MichaelMcAleer — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (MichaelMcAleer) [reply go.mod:3]: Resolving after internal discussion on this.

**Pull request merged**